desktop/core/src/desktop/js/apps/editor/components/ExecutableActionsKoBridge.vue

```diff
@@ -25,26 +25,47 @@
 </template>
 
 <script lang="ts">
+  import { defineComponent, PropType } from 'vue';
+
   import ExecutableActions from './ExecutableActions.vue';
   import SqlExecutable from 'apps/editor/execution/sqlExecutable';
   import SubscriptionTracker from 'components/utils/SubscriptionTracker';
-  import Vue from 'vue';
-  import Component from 'vue-class-component';
-  import { Prop } from 'vue-property-decorator';
-  import { wrap } from 'vue/webComponentWrapper';
 
-  @Component({
-    components: { ExecutableActions }
-  })
-  export default class ExecutableActionsKoBridge extends Vue {
-    @Prop()
-    executableObservable?: KnockoutObservable<SqlExecutable | undefined>;
-    @Prop()
-    beforeExecute?: () => Promise<void>;
+  import { wrap } from 'vue/webComponentWrap';
+
+  const ExecutableActionsKoBridge = defineComponent({
+    components: {
+      ExecutableActions
+    },
+
+    props: {
+      executableObservable: {
+        type: Object as PropType<KnockoutObservable<SqlExecutable | undefined>>,
+        default: undefined
+      },
+      beforeExecute: {
+        type: Object as PropType<() => Promise<void>>,
+        default: undefined
```

Comment:
If default: undefined type should be:

Object as PropType<(() => Promise<void>) | undefined>